src/main/java/bio/terra/JobService.java

```diff
@@ -0,0 +1,111 @@
+import bio.terra.stairway.FlightStatus;
+import bio.terra.stairway.Stairway;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class JobService {
+
+    private final Stairway stairway;
+
+    @Autowired
+    public JobService(
+            Stairway stairway
+    ) {
+        this.stairway = stairway;
+    }
+
+    public JobModel mapFlightStateToJobModel(FlightState flightState) {
+        FlightMap inputParameters = flightState.getInputParameters();
+        String description = inputParameters.get(JobMapKeys.DESCRIPTION.getKeyName(), String.class);
+        FlightStatus flightStatus = flightState.getFlightStatus(); // needs to be converted -- create a switch
+        JobModel.JobStatusEnum jobStatus = getJobStatus(flightStatus);
+        HttpStatus statusCode = inputParameters.get(JobMapKeys.STATUS_CODE.getKeyName(), HttpStatus.class);
+        String submittedDate = new SimpleDateFormat().format(flightState.getSubmitted());
+
+        String completedDate = null;
+        if (flightState.getCompleted().isPresent()) {
+            completedDate = new SimpleDateFormat().format(flightState.getCompleted());
+        }
+
+        JobModel jobModel = new JobModel()
+                .id(flightState.getFlightId())
+                .description(description)
+                .jobStatus(jobStatus)
+                .statusCode(statusCode.value())
+                .submitted(submittedDate)
+                .completed(completedDate);
+
+        return jobModel;
+    }
+
+    private JobModel.JobStatusEnum getJobStatus(FlightStatus flightStatus) {
+        switch (flightStatus) {
+            case ERROR: // TODO --why not FlightStatus.ERROR ?
+                return JobModel.JobStatusEnum.FAILED;
+            case FATAL:
+                return JobModel.JobStatusEnum.FAILED;
+            case RUNNING:
+                return JobModel.JobStatusEnum.RUNNING;
+            case SUCCESS:
+                return JobModel.JobStatusEnum.SUCCEEDED;
+        }
+        return JobModel.JobStatusEnum.FAILED;
+    }
+
+    public ResponseEntity<List<JobModel>> enumerateJobs(int offset, int limit) {
+        List<FlightState> flightStateList = stairway.getFlights(offset, limit);
+        List<JobModel> jobModelList = new ArrayList<>();
+        for (FlightState flightState : flightStateList) {
+            JobModel jobModel = mapFlightStateToJobModel(flightState);
+            jobModelList.add(jobModel);
+        }
+        return new ResponseEntity<>(jobModelList, HttpStatus.OK);
+    }
+
+    public ResponseEntity<JobModel> retrieveJob(String jobId) {
+        FlightState flightState = stairway.getFlightState(jobId);
+        String locationHeader  = String.format("/api/jobs/%s/result", jobId);
```

Comment:
We need a different location header depending on whether the job is completed or not.
If completed, then location is "/api/jobs/%s/result"
If not complete, then location should be "/api/jobs/%s"
That is, if it is not complete, we tell the client to call this method back. If it is complete, then we tell the client to hit the result endpoint.